test/location.coffee

```diff
@@ -123,7 +123,7 @@ test 'Verify locations in string interpolation (in "string", multiple interpolat
   eq c[2].last_column, 2
 
 test 'Verify locations in string interpolation (in "string", multiple interpolation and starting with line breaks)', ->
-  [a, b, c] = getMatchingTokens '"\n#{a}\nb\n#{c}"', 'a', '" b "', 'c'
+  [a, b, c] = getMatchingTokens '"\n#{a}\nb\n#{c}"', 'a', '\nb\n', 'c'
```

Comment:
`"\nb\n"` needs to still compile as `" b "`. To do otherwise would be a breaking change.

Reply:
@GeoffreyBooth yup so the difference is at the token level, since the original formatting is now preserved (rather than already transformed by the lexer before creating the tokens). Since these tests are checking against `CoffeeScript.tokens()` directly, they had to be updated to look for different `token[1]` values

Then it's in `StringLiteral::formatValue()` that the originally-formatted string `@originalValue` (which also doesn't have surrounding quotes included) gets transformed and quoted (eg here into `" b "`) and assigned to `@value`

So it still *compiles* to `" b "`, we have lots of tests for various string types that would be failing if that weren't the case